src/api.ts

```
@@ -1,11 +1,17 @@
 import { MovieData } from './types/MovieData';
-import { ResponseError } from './types/ReponseError';
+import { ResponseError } from './types/ResponseError';
 
-const API_URL = 'https://www.omdbapi.com/?apikey=your-key';
+const API_URL = 'https://www.omdbapi.com/?apikey=fbc18c26';
 
 export function getMovie(query: string): Promise<MovieData | ResponseError> {
   return fetch(`${API_URL}&t=${query}`)
-    .then(res => res.json())
+    .then(res => {
+      if (!res.ok) {
+        throw new Error('unexpected error');
+      }
+
+      return res.json();
+    })
     .catch(() => ({
```

The `.catch(() => ({ Response: 'False', Error: 'unexpected error' }))` swallows the real error. Prefer preserving the original error message when available, e.g. `.catch(err => ({ Response: 'False', Error: err?.message ?? 'unexpected error' }))`. This gives more actionable messages during debugging and aligns with the ResponseError shape the rest of the app expects.